client/client.go

```diff
@@ -48,6 +48,23 @@ var DefaultTransport CancelableTransport = &http.Transport{
 	TLSHandshakeTimeout: 10 * time.Second,
 }
 
+type EndpointSelectionMode int
+
+const (
+	// EndpointSelectionDefault is to pick an endpoint in a random manner.
+	EndpointSelectionDefault EndpointSelectionMode = iota
+
+	// EndpointSelectionPrioritizeLeader is to prioritize leader for reducing needless
+	// forward between follower and leader.
+	//
+	// This mode should be used with Client.AutoSync().
+	EndpointSelectionPrioritizeLeader
+
+	// EndpointSelectionPreference is to manually picks an endpoint
+	// with using Config.PreferenceIndex as an index.
+	EndpointSelectionPreference
```

Comment:
Actually I am not convinced that we need this. Let's remove this for now. We can consider adding this back when there is a clear need.